File: output/py/Medium/655-combinations.py
#https://leetcode.com/problems/combinations/
#Given two integers n and k, return all possible combinations of k numbers chosen from the range [1, n].
#You may return the answer in any order.


class Solution(object):
    def combine(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[List[int]]
        """ 
        res = []
        self.dfs(range(1, n + 1), k, 0, [], res)
        return res
        

    

File: output/java/Medium/655-combinations.java
/**https://leetcode.com/problems/combinations/ */
//Given two integers n and k, return all possible combinations of k numbers chosen from the range [1, n].
//You may return the answer in any order.


class Combinations {
    public List<List<Integer>> combine(int n, int k) {
        List<List<Integer>> res = new ArrayList<>();
        if(n == 0 || k == 0){
            return res;
        }
        helper(n, k, 1, new ArrayList<>(), res);
        return res;
        
        
        
    }
}

    